functions/src/messages/SendMessageHandler.ts

```diff
@@ -0,0 +1,141 @@
+	templateParameter: {
+		templatePath?: string;
+		translationNamespace?: string;
+		context: object;
+	};
+}
+
+export class SendMessageHandler {
+	readonly firestoreAdmin: FirestoreAdmin;
+
+	constructor(firestoreAdmin: FirestoreAdmin) {
+		this.firestoreAdmin = firestoreAdmin;
+	}
+	sendMessagesFunction = functions.https.onCall(
+		async (
+			{
+				communicationChannel,
+				targetAudience,
+				messageRecipients,
+				templateParameter: { templatePath, translationNamespace, context },
+			}: SendMessagesFunctionProps,
+			{ auth }
+		) => {
+			await this.firestoreAdmin.assertGlobalAdmin(auth?.token?.email);
+
+			let [successCount, skippedCount] = [0, 0];
+
+			if (communicationChannel === 'email') {
+				//TODO: Implement email util from Shared
+				console.log('Implementation not yet done');
+			} else if (communicationChannel === 'sms') {
+				[successCount, skippedCount] = await this.sendSmsMessage(targetAudience, messageRecipients, {
+					templatePath,
+					translationNamespace,
+					context,
+				});
+			}
+			return `Successfully sent out ${successCount} messages (${skippedCount} skipped)`;
+		}
+	);
+
+	sendSmsMessage = async (
+		targetAudience: string,
+		messageRecipients: Entity<User | Recipient>[],
+		templateParameter: {
+			templatePath?: string;
+			translationNamespace?: string;
+			context: object;
+		}
+	) => {
+		let [successCount, skippedCount] = [0, 0];
+
+		let messageRecipientPhone = '';
+		let messageRecipientLanguage = '';
+
+		for (const messageRecipientEntity of messageRecipients) {
+			try {
+				if (targetAudience === 'users') {
+					const user = messageRecipientEntity.values as User;
+					if (user.personal?.phone && user.language) {
+						messageRecipientLanguage = user.language;
+						messageRecipientPhone = user.personal.phone;
+					} else {
+						throw new Error('Either phone number or language missing for ' + messageRecipientEntity.id);
+					}
+				} else if (targetAudience === 'recipients') {
+					const recipient = messageRecipientEntity.values as Recipient;
+					if (recipient.mobile_money_phone && recipient.main_language) {
+						messageRecipientLanguage = recipient.main_language;
+						messageRecipientPhone = '+' + recipient.mobile_money_phone;
+					}
+				} else {
+					throw new Error('User type not supported');
+				}
+
+				const content = await renderTemplate({
+					language: messageRecipientLanguage,
+					translationNamespace: templateParameter.translationNamespace as string,
+					hbsTemplatePath: templateParameter.templatePath as string,
+					context: templateParameter.context as object,
+				});
+				const statusCallbackUrl = new URL('https://test.test');
+
+				// messageSid and messageStatus is required to add for Message Inbox
+				//const [messageSid, messageStatus] = await sendSms({
+				await sendSms({
+					messageRecipientPhone: messageRecipientPhone,
+					smsServiceId: TWILIO_SID,
+					smsServiceSecret: TWILIO_TOKEN,
+					statusCallbackUrl: statusCallbackUrl,
```

Comment:
Do we need to provide it? Can we leave the callback empty till we implemented it?
Maybe add a comment with a todo that at some point we want to implement the callback?